### PR TITLE
KEYCLOAK-17667 Client Policy - Executor : Only Accept Confidential Client

### PR DESCRIPTION
**Pull request by tnorimat**
This PR is for [KEYCLOAK-13933 Client Policies](https://issues.redhat.com/browse/KEYCLOAK-13933), also is the part of [FAPI-SIG](https://github.com/keycloak/kc-sig-fapi) activity. 

The details can be found in its [design document](https://github.com/keycloak/keycloak-community/blob/master/design/client-policies.md) and [KEYCLOAK-17667](https://issues.redhat.com/browse/KEYCLOAK-17667).

### CODE REVIEW

**Comment by tnorimat** [comment]
@mposolda The PR is for following the final version of FAPI 1.0 mentioned in page 8 of [FAPI-SIG 15th meeting agenda](https://github.com/keycloak/kc-sig-fapi/blob/master/FAPI-SIG/meetings/15th/presentations/FAPI-SIG_15th_MTG_agenda.pdf).
Could you or your team review this PR?

5.1.  Introduction of [the final version of FAPI 1.0 Advanced](https://openid.net/specs/openid-financial-api-part-2-1_0.html) states that this advanced security profile does not support public client.
This feature might be implemented by only allowing the confidential client on both authorization and token endpoint.

File: services/src/main/java/org/keycloak/services/clientpolicy/executor/ConfidentialClientAcceptExecutor.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+public class ConfidentialClientAcceptExecutor implements ClientPolicyExecutorProvider<ClientPolicyExecutorConfiguration> {
+
+    protected final KeycloakSession session;
+
+    public ConfidentialClientAcceptExecutor(KeycloakSession session) {
+        this.session = session;
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class Configuration {
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/services/clientpolicy/executor/ConfidentialClientAcceptExecutor.java:40]: It seems that this class is not needed at all - we can probably stick with the ClientPolicyExecutorConfiguration?

Just noticed that the same approach can be done on few other executor implementations (like for example SecureSessionEnforceExecutor) where the inner "Configuration" class can also be removed. It should also help to simplify some stuff in AbstractClientPoliciesTest, which can just use the generic ClientPolicyExecutorConfiguration for executors, which don't have any configuration options. WDYT?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/AbstractClientPoliciesTest.java [hunk 2]
```diff
@@ -889,6 +890,10 @@ protected Object createSecureSigningAlgorithmForSignedJwtEnforceExecutorConfig()
         return new SecureSigningAlgorithmForSignedJwtEnforceExecutor.Configuration();
     }
 
+    protected Object createConfidentialClientAcceptExecutorConfig() {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/AbstractClientPoliciesTest.java:893]: As mentioned above, this method can be probably removed (similar methods for other executors as well) and just use ClientPolicyExecutorConfiguration. WDYT?

**Review by mposolda — CHANGES_REQUESTED** [review]
@tnorimat Thanks for the PR! I have two inline comments. I am curious for your opinion (and eventually for remove also other "Configuration" class on some other executors in this or follow-up PR).

File: services/src/main/java/org/keycloak/services/clientpolicy/executor/ConfidentialClientAcceptExecutor.java (hunk 1 above)
Comment (tnorimat) [reply services/src/main/java/org/keycloak/services/clientpolicy/executor/ConfidentialClientAcceptExecutor.java]: @mposolda Yes, we need not prepare `Configuration` for the executor which do not have any configuration.
In `AbstractClientPoliciesTest`, I would like to remove the configuration methods for such executors.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/AbstractClientPoliciesTest.java (hunk 2 above)
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/AbstractClientPoliciesTest.java]: @mposolda I would like to remove such method as I've mentioned just above.

BTW, we already had such executors that do not have any configuration other than this `ConfidentialClientAcceptExecutor`, namely, `SecureRedirectUriEnforceExecutor`, `SecureResponseTypeExecutor`, `SecureSessionEnforceExecutor`, `SecureSigningAlgorithmEnforceExecutor` and  `SecureSigningAlgorithmForSignedJwtEnforceExecutor`.
I would like to create the follow-up JIRA ticket and try to do the same refactoring as this `ConfidentialClientAcceptExecutor`.

**Review by mposolda — APPROVED** [review]
@tnorimat Thanks for the update!

**Pull request merged**